Update Deploy 12 API key recommendation (#5199)
umbraco/UmbracoDocs: 12/umbraco-deploy/getting-started/install-configure.md

{% endhint %}

{% hint style="info" %}
In order to deploy content based on certain rich core and community property editors - including Nested Content, Multi URL Picker and Block List/Grid Editor - there is one further NuGet package to install: `Umbraco.Deploy.Contrib`.
{% endhint %}

> When Umbraco Deploy has been installed, to be able to use it in the project you will need to create and add configuration for an API key/secret.

Reviewer: ```suggestion
With Umbraco Deploy installed, to use it in the project you will need to create and add configuration for an API key/secret.
```

> For improved security, it is recommended to set the `ApiSecret` (instead of the `ApiKey`) setting to a **cryptographically random value of 64 bytes** (using Base64-encoding to get the string representation, resulting in a value of 88 characters). For versions prior to Deploy 12 or when not using the API secret setting, the recommendation is to set the `ApiKey` to a randomly generated string of 64 characters.

Reviewer: ```suggestion
For improved security, it is recommended to set the `ApiSecret` (instead of the `ApiKey`) setting to a **cryptographically random value of 64 bytes**. Using Base64-encoding to get the string representation, will result in a value of 88 characters. For versions prior to Deploy 12 or when not using the API secret setting, the recommendation is to set the `ApiKey` to a randomly generated string of 64 characters.
```

When Umbraco Deploy has been installed, to be able to use it in the project you will need to create and add configuration for an API key.

> The recommended API key value is a **Base64-encoded string of 64 randomly generated bytes** (which will result in a value of 88 characters), although to maintain backwards compatibility the minimum requirement is 10 characters.

Reviewer 2: ```suggestion
The recommended API key value is a **Base64-encoded string of 64 randomly generated bytes**, resulting in a value of 88 characters. In order to maintain backward compatibility the minimum requirement is 10 characters.
```
